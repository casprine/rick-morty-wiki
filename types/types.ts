export interface Character {
  id: number;
  name: string;
  species: string;
  image: string;
  status: 'alive' | 'dead' | 'unknown';
  location: Location;
  gender: string;
}

export interface CharacterResponse {
  info: RequestInfo;
  results: Character[];
}

export interface RequestInfo {
  count: number;
  pages: number;
  next: string;
  prev: string | null;
}

export interface Location {
  name: string;
  url: string;
}

export interface Episode {
  id: number;
  name: string;
  air_date: string;
  episode: string;
  characters: Character[];
  url: string;
  created: string;
}

export interface GetCharacterParams {
  status?: string;
  species?: string;
  pageNumber?: number;
  gender?: string;
}

export type FilterType = 'status' | 'gender' | 'species';
