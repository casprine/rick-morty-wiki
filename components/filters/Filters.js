"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
exports.__esModule = true;
exports.Filters = void 0;
var react_select_1 = require("react-select");
var stitches_1 = require("@/stitches");
var helpers_1 = require("lib/helpers");
var filterOptions = [
    {
        label: 'status',
        values: ['alive', 'dead', 'unknown']
    },
    {
        label: 'species',
        values: [
            'alien',
            'human',
            'poopybutthole',
            'humanoid',
            'mythological',
            'unknown',
            'animal',
            'disease',
            'robot',
            'cronenberg',
            'planet',
        ]
    },
    {
        label: 'gender',
        values: ['female', 'male', 'genderless', 'unknown']
    },
];
var Filter = function (_a) {
    var onFilterClick = _a.onFilterClick, label = _a.label, values = _a.values;
    return (<FilterContainer>
      <p className="label">{label}</p>
      <react_select_1["default"] styles={selectStyles} onChange={function (_a) {
        var value = _a.value;
        return onFilterClick(label, value);
    }} options={values.map(function (v) {
            return {
                value: v,
                label: (0, helpers_1.capitalize)(v)
            };
        })}/>
    </FilterContainer>);
};
var Filters = function (_a) {
    var onClear = _a.onClear, onFilterChange = _a.onFilterChange;
    return (<Container className="flex">
      <div className="actions flex">
        {filterOptions.map(function (filter, index) {
            return <Filter {...filter} key={index} onFilterClick={onFilterChange}/>;
        })}

        <button onClick={onClear}>Clear All</button>
      </div>
    </Container>);
};
exports.Filters = Filters;
var Container = (0, stitches_1.styled)('div', {
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 30,
    '*': {
        outline: '1px dotted red'
    },
    '.header': {
        display: 'flex',
        alignItems: 'center',
        width: '100%',
        justifyContent: 'space-between'
    },
    '.actions': {
        justifyContent: 'center',
        alignItems: 'center'
    },
    button: {
        border: 0,
        backgroundColor: 'transparent',
        color: '$blue',
        cursor: 'pointer',
        fontSize: 16
    }
});
var FilterContainer = (0, stitches_1.styled)('div', {
    marginTop: 20,
    marginLeft: '5px',
    marginRight: '5px',
    '.label': {
        textTransform: 'uppercase',
        fontSize: 14,
        fontWeight: 500,
        marginBottom: 5
    }
});
var selectStyles = {
    container: function (styles) { return (__assign(__assign({}, styles), { width: 200 })); },
    input: function (styles) { return (__assign(__assign({}, styles), { fontSize: 15, height: 25 })); },
    placeholder: function (styles) { return (__assign(__assign({}, styles), { fontSize: 15 })); },
    singleValue: function (styles) { return (__assign(__assign({}, styles), { fontSize: 15 })); },
    option: function (styles) { return (__assign(__assign({}, styles), { fontSize: 15 })); }
};
