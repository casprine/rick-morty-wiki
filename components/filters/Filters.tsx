import { FunctionComponent } from 'react';
import Select, { StylesConfig } from 'react-select';

import { styled } from '@/stitches';
import { FilterType } from '@/types';
import { capitalize } from 'lib/helpers';

type filterOption = {
  label: string;
  values: string[];
};

const filterOptions: filterOption[] = [
  {
    label: 'status',
    values: ['alive', 'dead', 'unknown'],
  },

  {
    label: 'species',
    values: [
      'alien',
      'human',
      'poopybutthole',
      'humanoid',
      'mythological',
      'unknown',
      'animal',
      'disease',
      'robot',
      'cronenberg',
      'planet',
    ],
  },

  {
    label: 'gender',
    values: ['female', 'male', 'genderless', 'unknown'],
  },
];

interface FiltersProps {
  // updatePageNumber?: (pageNumber: number) => void;
  onClear: () => void;
  onFilterChange: (type: FilterType, value: string) => void;
}

interface FilterProps {
  label: string;
  onFilterClick: (label: string, filter: string) => void;
  values: string[];
}

const Filter: FunctionComponent<FilterProps> = ({ onFilterClick, label, values }) => {
  return (
    <FilterContainer>
      <p className="label">{label}</p>
      <Select
        styles={selectStyles}
        onChange={({ value }: { value: string }) => onFilterClick(label, value)}
        options={values.map((v: string) => {
          return {
            value: v,
            label: capitalize(v),
          };
        })}
      />
    </FilterContainer>
  );
};

export const Filters: FunctionComponent<FiltersProps> = ({ onClear, onFilterChange }) => {
  return (
    <Container className="flex">
      <div className="actions flex">
        {filterOptions.map((filter: filterOption, index: number) => {
          return <Filter {...filter} key={index} onFilterClick={onFilterChange} />;
        })}

        <button onClick={onClear}>Clear All</button>
      </div>
    </Container>
  );
};

const Container = styled('div', {
  justifyContent: 'center',
  alignItems: 'center',
  marginBottom: 30,

  '*': {
    outline: '1px dotted red',
  },

  '.header': {
    display: 'flex',
    alignItems: 'center',
    width: '100%',
    justifyContent: 'space-between',
  },

  '.actions': {
    justifyContent: 'center',
    alignItems: 'center',
  },

  button: {
    border: 0,
    backgroundColor: 'transparent',
    color: '$blue',
    cursor: 'pointer',
    fontSize: 16,
  },
});

const FilterContainer = styled('div', {
  marginTop: 20,
  marginLeft: '5px',
  marginRight: '5px',

  '.label': {
    textTransform: 'uppercase',
    fontSize: 14,
    fontWeight: 500,
    marginBottom: 5,
  },
});

const selectStyles: StylesConfig<{}> = {
  container: (styles) => ({
    ...styles,
    width: 200,
  }),

  input: (styles) => ({
    ...styles,
    fontSize: 15,
    height: 25,
  }),

  placeholder: (styles) => ({
    ...styles,
    fontSize: 15,
  }),
  singleValue: (styles) => ({
    ...styles,
    fontSize: 15,
  }),
  option: (styles) => ({
    ...styles,
    fontSize: 15,
  }),
};
