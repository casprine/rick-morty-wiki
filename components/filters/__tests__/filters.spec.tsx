import { Filters } from '../Filters';
import { render } from '@testing-library/react';
import { FilterType } from '@/types';

const initialFilters: Record<FilterType, string> = {
  status: '',
  gender: '',
  species: '',
};

test('renders a filter with empty state', () => {
  const component = render(<Filters onClear={jest.fn()} onFilterChange={jest.fn()} filterValues={initialFilters} />);
  expect(component).toMatchSnapshot();
});
