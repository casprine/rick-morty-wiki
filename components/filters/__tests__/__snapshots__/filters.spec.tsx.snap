// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a filter with empty state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="c-chCaDx flex"
      >
        <div
          class="c-hiiSKh"
        >
          <p
            class="label"
          >
            status
          </p>
          <div
            class=" css-1ktriz-container"
            id="filter-select"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-filter-select-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-1s2u09g-control"
            >
              <div
                class=" css-319lph-ValueContainer"
              >
                <div
                  class=" css-l2btf2-placeholder"
                  id="react-select-filter-select-placeholder"
                >
                  Select a status
                </div>
                <div
                  class=" css-s6tp6q-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-describedby="react-select-filter-select-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-filter-select-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c-hiiSKh"
        >
          <p
            class="label"
          >
            species
          </p>
          <div
            class=" css-1ktriz-container"
            id="filter-select"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-filter-select-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-1s2u09g-control"
            >
              <div
                class=" css-319lph-ValueContainer"
              >
                <div
                  class=" css-l2btf2-placeholder"
                  id="react-select-filter-select-placeholder"
                >
                  Select a species
                </div>
                <div
                  class=" css-s6tp6q-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-describedby="react-select-filter-select-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-filter-select-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c-hiiSKh"
        >
          <p
            class="label"
          >
            gender
          </p>
          <div
            class=" css-1ktriz-container"
            id="filter-select"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-filter-select-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-1s2u09g-control"
            >
              <div
                class=" css-319lph-ValueContainer"
              >
                <div
                  class=" css-l2btf2-placeholder"
                  id="react-select-filter-select-placeholder"
                >
                  Select a gender
                </div>
                <div
                  class=" css-s6tp6q-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-describedby="react-select-filter-select-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-filter-select-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button>
          Clear All
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="c-chCaDx flex"
    >
      <div
        class="c-hiiSKh"
      >
        <p
          class="label"
        >
          status
        </p>
        <div
          class=" css-1ktriz-container"
          id="filter-select"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-filter-select-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1s2u09g-control"
          >
            <div
              class=" css-319lph-ValueContainer"
            >
              <div
                class=" css-l2btf2-placeholder"
                id="react-select-filter-select-placeholder"
              >
                Select a status
              </div>
              <div
                class=" css-s6tp6q-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-filter-select-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-filter-select-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c-hiiSKh"
      >
        <p
          class="label"
        >
          species
        </p>
        <div
          class=" css-1ktriz-container"
          id="filter-select"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-filter-select-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1s2u09g-control"
          >
            <div
              class=" css-319lph-ValueContainer"
            >
              <div
                class=" css-l2btf2-placeholder"
                id="react-select-filter-select-placeholder"
              >
                Select a species
              </div>
              <div
                class=" css-s6tp6q-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-filter-select-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-filter-select-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c-hiiSKh"
      >
        <p
          class="label"
        >
          gender
        </p>
        <div
          class=" css-1ktriz-container"
          id="filter-select"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-filter-select-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1s2u09g-control"
          >
            <div
              class=" css-319lph-ValueContainer"
            >
              <div
                class=" css-l2btf2-placeholder"
                id="react-select-filter-select-placeholder"
              >
                Select a gender
              </div>
              <div
                class=" css-s6tp6q-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-filter-select-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-filter-select-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button>
        Clear All
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
